source/ns_cmdline.c

```diff
@@ -1707,9 +1707,13 @@ int help_command(int argc, char *argv[])
 int history_command(int argc, char *argv[])
 {
     if (argc == 1) {
-        cmd_printf("History [%i/%i]:\r\n", (int)ns_list_count(&cmd.history_list), cmd.history_max_count);
+        int history_size = (int)ns_list_count(&cmd.history_list) - 1;
+        cmd_printf("History [%i/%i]:\r\n", history_size-1, cmd.history_max_count);
```

Comment:
1. Max: `History [30/32]`
2. 
```
>help
>help
>help
>history:
History [1/32]:
[0]: help
[1]: history
```

Reply:
:) not. Seems that it needs more tuning

Reply:
in fact - it is expected. By default it does not store duplicates, see:
https://github.com/ARMmbed/mbed-client-cli/blob/master/source/ns_cmdline.c#L1351
maybe better to switch it off by default..

Reply:
actually, ubuntu terminal treats command history the same way  --> `bool noduplicate = true;`